src/config.js

```diff
@@ -276,9 +276,27 @@ export default class Config {
       networkConcurrency: this.networkConcurrency,
       networkTimeout: this.networkTimeout,
     });
-    this._cacheRootFolder = String(
-      opts.cacheFolder || this.getOption('cache-folder', true) || constants.MODULE_CACHE_DIRECTORY,
-    );
+
+    let cacheRootFolder = opts.cacheFolder || this.getOption('cache-folder', true);
+
+    for (let t = 0; t < constants.PREFERRED_MODULE_CACHE_DIRECTORIES.length && !cacheRootFolder; ++t) {
+      const tentativeCacheFolder = constants.PREFERRED_MODULE_CACHE_DIRECTORIES[t];
+
+      try {
+        await fs.mkdirp(tentativeCacheFolder);
+        await fs.writeFile(path.join(tentativeCacheFolder, constants.WTEST_FILENAME), `testing write access`);
```

Comment:
After discussing this with @BYK we came to the conclusion that `--cache-folder` would keep its current semantic (and would crash if the directory is not writable) since it would break the commonly assumed contract, potentially exposing the application to vulnerabilities (the `/tmp` folder could be world-readable whereas the `--cache-folder` wouldn't). However, we'll also add a `--preferred-cache-folder` option with less restrictions. Will even make it easier to test the feature :)